beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -3630,14 +3636,25 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
             .map(|ep| ep.block_hash)
             .unwrap_or_else(Hash256::zero);
 
-        execution_layer
+        let forkchoice_updated_response = execution_layer
             .notify_forkchoice_updated(
                 head_execution_block_hash,
                 finalized_execution_block_hash,
                 None,
             )
             .await
-            .map_err(Error::ExecutionForkChoiceUpdateFailed)
+            .map_err(Error::ExecutionForkChoiceUpdateFailed);
+
+        match forkchoice_updated_response {
+            Ok((status, latest_valid_hash)) => match status {
+                PayloadStatusV1Status::Valid | PayloadStatusV1Status::Syncing => Ok(()),
+                PayloadStatusV1Status::Invalid => Err(
+                    BeaconChainError::ExecutionForkChoiceUpdateInvalidHead(latest_valid_hash),
+                ),
+                status => panic!("Unrecognized status from forkchoice_updated: {:?}", status),
```

Comment:
This would never occur as the code is currently written. Only if the code was changed incorrectly

Reply:
Even if that's the case, we still don't want leave `panic`s around for future developers to hit if they make a mistake. Panics don't pass errors downstream and therefore the caller (probably the network or the API in this case) can't do any error handling of it's own.

It should be trivial to handle the `Err` here. If we *were* to leave a panic around here because there's no other option, then I'd use an `unreachable!` and provide very clear comments as to why we won't hit this panic.